src/coreclr/vm/method.cpp

```diff
@@ -4124,6 +4072,54 @@ void MethodDesc::PrepareForUseAsADependencyOfANativeImageWorker()
     EX_END_CATCH(RethrowTerminalExceptions);
     _ASSERTE(HaveValueTypeParametersBeenWalked());
 }
+
+static void CheckForEquivalenceAndLoadType(Module *pModule, mdToken token, Module *pDefModule, mdToken defToken, const SigParser *ptr, SigTypeContext *pTypeContext, void *pData)
```

Comment:
This function no longer behaves like the copy of `CheckForEquivalenceAndLoadType` that is in methodtable.cpp. In particular, it unconditionally loads the type unlike the code in methodtable.cpp which is carefully constructed to not trigger additional loads. Also given that both of these paths set the HaveValueTypeParametersBeenWalked flag, I think we may not have a fully functional implementation. In particular, imagine the case where a virtual function is loaded, and then later `ldftn` is used on it. I believe that the HaveValueTypeParametersBeenWalked flag will already be set, but we may not have loaded all the parameters.

Reply:
Just to be clear with respect to the `CheckForEquivalenceAndFullyLoadType` in `methodtable.cpp` &mdash; I assume you mean `CheckForEquivalenceAndFullyLoadType` because I can't find `CheckForEquivalenceAndLoadType` &mdash; the logic was different even before I changed anything. The one in `methodtable.cpp` has extra logic that `CheckForEquivalenceAndLoadType` never had. Specifically, a call to `DoFullyLoad()`

https://github.com/dotnet/runtime/blob/27226f631ff5e6019b7da78d350636a6e964aa3c/src/coreclr/vm/methodtable.cpp#L3908-L3916

whereas `CheckForEquivalenceAndLoadType` never had that call:

https://github.com/dotnet/runtime/blob/27226f631ff5e6019b7da78d350636a6e964aa3c/src/coreclr/vm/method.cpp#L4018-L4023

Reply:
Spoke offline. This seems okay, but a new test is warranted.